src/zoid/buttons/util.js

```diff
@@ -82,9 +82,35 @@ export function applePaySession() : ?ApplePaySessionConfigRequest {
 
         return (version, request) => {
             const session = new window.ApplePaySession(version, request);
+            const listeners = {};
+
+            session.onvalidatemerchant = (e) => {
+                listeners.validatemerchant(e.validationURL);
+            };
+
+            session.onpaymentmethodselected = () => {
```

Comment:
This approach works for me, I guess this also means we can decide exactly which parts of the `event` to map and pass back. I think it's probably rare we add new events in future.